staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -950,6 +1086,17 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 	storageMap2[crd.UID] = ret
 	r.customStorage.Store(storageMap2)
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StorageVersionAPI) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.APIServerIdentity) {
+		return ret, nil
+	}
+
```

Comment:
// spawn storage version update in background and use channels to make handlers wait